portal/src/main/webapp/js/src/patient-view/util/ClinicalAttributesUtil.js

```diff
@@ -0,0 +1,203 @@
+      'n/a',
+      '[unkown]',
+      '[not submitted]',
+      '[not evaluated]',
+      '[not applicable]',
+      '[not available]',
+      '[undefined]'
+    ];
+
+    for (var key in clinicalData) {
+      var cleanValue;
+      var value;
+
+      value = clinicalData[key];
+
+      // Remove null values
+      if (NULL_VALUES.indexOf(value.toLowerCase()) > -1) {
+        delete cleanClinicalData[key];
+      } else {
+        // Change values for certain attributes, e.g. rounding
+        switch (key) {
+          case 'OS_MONTHS':
+          case 'DFS_MONTHS':
+            if ($.isNumeric(value)) {
+              value = Math.round(value);
+            }
+            cleanClinicalData[key] = value;
+            break;
+        }
+      }
+    }
+    return cleanClinicalData;
+  }
+
+  /**
+   * Get first key found in object. Otherwise return null.
+   * @param {object} object - object with key/value pairs
+   * @param {array} keys - array of keys
+   */
+  function getFirstKeyFound(object, keys) {
+      if (!object) {
+        return null;
+      }
+
+      for (var i=0; i< keys.length; i++) {
+          var value = object[keys[i]];
+          if (typeof value !== 'undefined' && value !== null) {
+            return value;
+          }
+      }
+      return null;
+  }
+
+
+  /**
+   * Derive clinical attributes from existing clinical attributes .e.g. age based
+   * on a date of birth. TODO: Now only includes a funky hack to keep current
+   * derived clinical attributes working.
+   * @param {object} clinicalData - key/value pairs of clinical data
+   */
+  function derive(clinicalData) {
+    var derivedClinicalAttributes = $.extend({}, clinicalData);
+    
+    /**
+     * TODO: Pretty funky function to get a normalized case type. This should
+     * probably also be a clinical attribute with a restricted vocabulary. Once
+     * the database has been changed to include normalized case types, this
+     * function should be removed.
+     * @param {object} clinicalData - key/value pairs of clinical data
+     * @param {string} caseTypeAttrs - TUMOR_TYPE or SAMPLE_TYPE value to normalize
+     */
+    function normalizedCaseType(clinicalData, caseTypeAttrs) {
+      var caseTypeNormalized = null;
+      var caseType;
+      var caseTypeLower;
+      var attr;
+      var i;
+
+      for (i = 0; i < caseTypeAttrs.length; i++) {
+        caseType = clinicalData[caseTypeAttrs[i]];
+
+        if (!cbio.util.checkNullOrUndefined(caseType)) {
+          caseTypeLower = caseType.toLowerCase();
+
+          if (caseTypeLower.indexOf('metasta') >= 0) {
+            caseTypeNormalized = 'metastasis';
+          } else if (caseTypeLower.indexOf('recurr') >= 0) {
+            caseTypeNormalized = 'recurrence';
+          } else if (caseTypeLower.indexOf('progr') >= 0) {
+            caseTypeNormalized = 'progressed';
+          } else if (caseTypeLower.indexOf('prim') >= 0 || 
+                    caseTypeLower.indexOf('prim') >= 0) {
```

Comment:
But am I misreading it if I think the two sides of the `||` clause are identical?